File: 2799.py
class Solution:
    def countCompleteSubarrays(self, nums: List[int]) -> int:
        size = len(set(nums))

        result = 0
        left = 0
        freq = defaultdict(int)

        for right in range(len(nums)):
            freq[nums[right]] += 1

            while len(freq) == size:
                result += len(nums) - right
                freq[nums[left]] -= 1
                if freq[nums[left]] == 0:
                    del freq[nums[left]]
                left += 1
        
        return result

File: 2799.java
class Solution {
    public int countCompleteSubarrays(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num : nums){
            set.add(num);
        }
        int size = set.size();

        int left = 0;
        int result = 0;
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int right = 0; right < nums.length; right++) {
            map.put(nums[right], map.getOrDefault(nums[right], 0) + 1);

            while(map.size() == size){
                result += nums.length - right;

                map.put(nums[left], map.get(nums[left]) - 1);
                if(map.get(nums[left]) == 0){
                    map.remove(nums[left]);
                }
                left++;
            }
        }
        return result;
    }
}
